### PR TITLE
[exporter/Logzioexporter] logs support

### PR DESCRIPTION
**Pull request by yotamloe**
**Description:** <Describe what has changed.>
logzioexporter update:
  * Add support for logs pipeline
  * Remove `logzio-go` and `jaeger-logzio` dependencies
  * Add `logzioSpan`,`logzioService`,`jsonLog` in object directiry
  * Add support for [exporter helper](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) configuration options (queue, retry)
  * Generic `export` method for traces and logs using HTTP client
  *  Warning logs for soon to be deprecated config options:
       * `custom_endpoint`
       * `drain_interval`
       *  `queue_capacity`
       * `queue_max_length`
  

**Testing:** <Describe what testing was performed and which tests were added.>
Added tests for logs pipeline and for logzio objects in the following files:
* `exporter_test.go`
* `config_test.go`
* `factory_test.go`
* `jsonlog_test.go`
* `logziospan_test.go`

Passing all unit tests:
coverage: 82% of statements
**Documentation:**
* Added documentation for `exporterhelper` configuration option
* Removed deprecated configuration options (custom_endpoint, drain_interval, queue_capacity, queue_max_length)
* Added examples for traces and logs pipeline configuration

### CODE REVIEW

**Comment by dashpole** [comment]
You might want to consider gating the configuration change with a feature-gate.  I did this in the googlecloud exporter when we made changes to our config.  See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/factory.go#L55 for an example of using different configurations based on a feature-gate.

**Comment by yotamloe** [comment]
Thanks for the advice @dashpole, I'll take a look at that :) 
Regarding the code owners, I'll add the relevant people, should I do it in a different PR?

**Comment by dashpole** [comment]
yes, please add owners in a separate PR.  If you could have one of the other codeowners review this PR, that would be helpful.

**Comment by dashpole** [comment]
Ah, sorry.  I searched for logsio...

**Comment by dashpole** [comment]
You wouldn't need to feature-gate the addition of logs support -- just the removal of config options from your stable configuration.

You just have to abide by [this](https://github.com/open-telemetry/opentelemetry-collector/blob/19cc093b42cb53ec0c7bfd2b9579619e8d89891d/README.md#stable), since your component is stable:

```Breaking changes, including configuration options and the component's output are not expected to happen without prior notice, unless under special circumstances.```

A feature-gate, which is initially disabled by default, is one way to provide prior notice of removal of those options, since users can temporarily (during beta) re-enable them.  But if you have a different way you'd like to provide prior notice (e.g. a warning log when using one of the options you plan to remove), that would also be acceptable.  You should have at least one release where it is in alpha and one in beta, after which you can remove the feature gate.

**Comment by yotamloe** [comment]
Ok, that makes sense.
So if ill keep the legacy options for the next releases, map the values to the new config options when possible (`custom_endpoint` -> `endpoint` for example) and add a warning log for upcoming deprecation and the alternative config options (when the legacy option is used), would that be acceptable?

**Comment by yotamloe** [comment]
Hi @dashpole, I hope you had a nice weekend.
I added the warning messages and option mapping in the latest [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10821/commits/5258a90faa982db28d94bc34cb32486a4adf8ba3)

File: exporter/logzioexporter/config.go [hunk 1]
```diff
@@ -17,24 +17,53 @@ package logzioexporter // import "github.com/open-telemetry/opentelemetry-collec
...
+	if config.Token == "" {
 		return errors.New("`account_token` not specified")
 	}
 	return nil
 }
+
+// CheckAndWarnDeprecatedOptions Is checking for soon deprecated configuration options (queue_max_length, queue_capacity, drain_interval, custom_endpoint) log a warning message and map to the relevant updated option
+func (config *Config) CheckAndWarnDeprecatedOptions(logger hclog.Logger) {
+	if config.QueueCapacity != 0 {
+		logger.Warn("You are using `queue_capacity` option that will be deprecated in the next releases, use exporter helper configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```
Comment (dashpole) [inline exporter/logzioexporter/config.go:51]: nit
```suggestion
		logger.Warn("You are using  the deprecated`queue_capacity` option that will be removed in the next release; use exporter helper configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```

File: exporter/logzioexporter/config.go [hunk 2]
```diff
@@ -17,24 +17,53 @@ package logzioexporter // import "github.com/open-telemetry/opentelemetry-collec
...
-	if c.TracesToken == "" {
+func (config *Config) Validate() error {
+	if config.Token == "" {
 		return errors.New("`account_token` not specified")
 	}
 	return nil
 }
+
+// CheckAndWarnDeprecatedOptions Is checking for soon deprecated configuration options (queue_max_length, queue_capacity, drain_interval, custom_endpoint) log a warning message and map to the relevant updated option
+func (config *Config) CheckAndWarnDeprecatedOptions(logger hclog.Logger) {
```
Comment (dashpole) [inline exporter/logzioexporter/config.go:49]: This function probably doesn't need to be exported.

File: exporter/logzioexporter/config.go [hunk 3]
```diff
@@ -17,24 +17,53 @@ package logzioexporter // import "github.com/open-telemetry/opentelemetry-collec
...
+	if config.QueueCapacity != 0 {
+		logger.Warn("You are using `queue_capacity` option that will be deprecated in the next releases, use exporter helper configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
+	}
+	// Warn and map queue_max_length -> QueueSettings.QueueSize
+	if config.QueueMaxLength != 0 {
+		logger.Warn("You are using `queue_max_length` option that will be deprecated in the next releases, use exporter helper `queue_size` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```
Comment (dashpole) [inline exporter/logzioexporter/config.go:55]:
```suggestion
		logger.Warn("You are using the deprecated `queue_max_length` option that will be removed in the next release; use exporter helper `queue_size` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md")
```

File: exporter/logzioexporter/config.go [hunk 4]
```diff
@@ -17,24 +17,53 @@ package logzioexporter // import "github.com/open-telemetry/opentelemetry-collec
...
+		config.QueueSettings.Enabled = true
+		logger.Warn("Mapping `queue_max_length` -> `QueueSettings.QueueSize`")
+		config.QueueSettings.QueueSize = config.QueueMaxLength
+	}
+	if config.DrainInterval != 0 {
+		logger.Warn("You are using `drain_interval` option that will be deprecated in the next releases, use batch processor `timeout` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor#batch-processor")
```
Comment (dashpole) [inline exporter/logzioexporter/config.go:61]:
```suggestion
		logger.Warn("You are using the deprecated `drain_interval` option that will be removed in the next release; use batch processor `timeout` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor#batch-processor")
```

File: exporter/logzioexporter/config.go [hunk 5]
```diff
@@ -17,24 +17,53 @@ package logzioexporter // import "github.com/open-telemetry/opentelemetry-collec
...
+	}
+	// Warn and map CustomEndpoint -> Endpoint
+	if config.CustomEndpoint != "" {
+		logger.Warn("You are using `custom_endpoint` option that will be deprecated in the next releases, please use `endpoint` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp")
```
Comment (dashpole) [inline exporter/logzioexporter/config.go:65]:
```suggestion
		logger.Warn("You are using the deprecated `custom_endpoint` option that will be removed in the next release; please use `endpoint` configuration instead: https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp")
```

**Review by dashpole — COMMENTED** [review]
I've reviewed the config + readme changes, which look good.  I'll defer to one of the codeowners (who are hopefully more familiar with logzio) to review the log exporter implementation.

**Comment by yotamloe** [comment]
I cant add reviewers. Can you please add @Doron-Bargo as a reviewer?

**Comment by dashpole** [comment]
I can't, since they don't appear to be org members.  Its fine though; they can still review.

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Comment by yotamloe** [comment]
Hi @dashpole.
Anything else we need to move forward and approve this PR?

File: exporter/logzioexporter/README.md [hunk 6]
```diff
@@ -83,6 +165,3 @@ service:
     logs:
       level: debug #activate debug mode
 ```
-
-[stable]:https://github.com/open-telemetry/opentelemetry-collector#stable
```
Comment (dashpole) [inline exporter/logzioexporter/README.md:87]: removing this breaks the link above (I think).  Can you add it back, and also add a link to beta (so the logs link works correctly).

**Review by dashpole — APPROVED** [review]

**Comment by yotamloe** [comment]
Hi @dashpole The lint test is failing due to an [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/runs/7029019592?check_suite_focus=true) in `tanzuobservabilityexporter` component, `logzioexporter` is passing the tests, how can we resolve this?

**Comment by mx-psi** [comment]
@yotamloe can you fix the merge conflicts?

File: CHANGELOG.md [hunk 7]
```diff
@@ -45,6 +45,7 @@
 - `datadogexporter`: Deprecate `Sanitize` method of `Config` struct (#8829)
 - `observiqexporter`: Deprecate the observiq exporter (#10977)
 - `honeycombexporter`: Deprecate honeycomb exporter (#10318)
+- `logzioexporter`: Announcing `custom_endpoint`, `drain_interval`, `queue_capacity`, `queue_max_length` configuration options will be deprecated in upcoming releases (#10821)
```
Comment (mx-psi) [inline CHANGELOG.md:48]: This should be moved to the 'Unreleased' section

File: CHANGELOG.md [hunk 8]
```diff
@@ -54,6 +55,7 @@
 
 ### 💡 Enhancements 💡
 
+- `logzioexporter`: Add support for logs pipeline and support for exporterhelper (#10821)
```
Comment (mx-psi) [inline CHANGELOG.md:58]: This should be moved to the 'Unreleased' section

File: exporter/logzioexporter/go.mod [hunk 9]
```diff
@@ -69,17 +52,9 @@ require (
...
-	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
 // see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4433
 exclude github.com/StackExchange/wmi v1.2.0
-
-replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger => ../../pkg/translator/jaeger
-
-replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal => ../../internal/coreinternal
```
Comment (mx-psi) [inline exporter/logzioexporter/go.mod:85]: These should not be removed, the component still uses both modules

File: exporter/logzioexporter/logzioservice.go [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+// LogzioService type, for query purposes
+type LogzioService struct {
+	OperationName string `json:"operationName"`
+	ServiceName   string `json:"serviceName"`
+	Type          string `json:"type"`
+}
+
+// NewLogzioService creates a new logzio service from a span
+func NewLogzioService(span *model.Span) LogzioService {
```
Comment (mx-psi) [inline exporter/logzioexporter/logzioservice.go:34]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+	var values []string
+	for i := range tags {
+		values = append(values, tags[i].VStr)
+	}
+	return values
+}
+
+// TransformToLogzioSpanBytes receives a Jaeger span, converts it to logzio span and returns it as a byte array.
+// The main differences between Jaeger span and logzio span are arrays which are represented as maps
+func TransformToLogzioSpanBytes(span *model.Span) ([]byte, error) {
```
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:58]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+const (
+	spanLogType = "jaegerSpan"
+	// TagDotReplacementCharacter state which character should replace the dot in es
+	TagDotReplacementCharacter = "@"
+)
+
+// LogzioSpan is same as esSpan with a few different json field names and an addition on type field.
+type LogzioSpan struct {
```
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:31]: Does this need to be public?

File: exporter/logzioexporter/logziospan.go [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+	"encoding/json"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/dbmodel"
+)
+
+const (
+	spanLogType = "jaegerSpan"
+	// TagDotReplacementCharacter state which character should replace the dot in es
+	TagDotReplacementCharacter = "@"
```
Comment (mx-psi) [inline exporter/logzioexporter/logziospan.go:27]: Does this need to be public?

File: exporter/logzioexporter/jsonlog.go [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+	case pcommon.ValueTypeEmpty:
+		return nil
+	default:
+		logger.Debug("Unhandled value type", zap.String("type", value.Type().String()))
+		return value
+	}
+}
+
+// ConvertLogRecordToJSON Takes `plog.LogRecord` and `pcommon.Resource` input, outputs byte array that represents the log record as json string
+func ConvertLogRecordToJSON(log plog.LogRecord, resource pcommon.Resource, logger hclog.Logger) map[string]interface{} {
```
Comment (mx-psi) [inline exporter/logzioexporter/jsonlog.go:57]: Does this need to be public?

**Review by mx-psi — COMMENTED** [review]
This PR adds some new public API, is it intentional? In general, functions, methods and structs should be made private so that we can change the internals of the exporter without making breaking changes.

File: exporter/logzioexporter/logzioservice.go (hunk 10 above)
Comment (yotamloe) [reply exporter/logzioexporter/logzioservice.go:34]: I'll make this private

File: exporter/logzioexporter/logziospan.go (hunk 11 above)
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:58]: I'll make this private

File: exporter/logzioexporter/logziospan.go (hunk 12 above)
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:31]: I'll make this private

File: exporter/logzioexporter/logziospan.go (hunk 13 above)
Comment (yotamloe) [reply exporter/logzioexporter/logziospan.go:27]: I'll make this private

File: exporter/logzioexporter/factory.go [hunk 15]
```diff
@@ -28,27 +37,74 @@ func NewFactory() component.ExporterFactory {
...
+	exporterConfig := cfg.(*Config)
+	return newLogzioTracesExporter(exporterConfig, params)
 }
 
-func createMetricsExporter(_ context.Context, params component.ExporterCreateSettings, cfg config.Exporter) (component.MetricsExporter, error) {
-	config := cfg.(*Config)
-	return newLogzioMetricsExporter(config, params)
+func CreateLogsExporter(_ context.Context, params component.ExporterCreateSettings, cfg config.Exporter) (component.LogsExporter, error) {
+	exporterConfig := cfg.(*Config)
+	return newLogzioLogsExporter(exporterConfig, params)
```
Comment (mx-psi) [inline exporter/logzioexporter/factory.go:109]: Missed these, sorry, if possible we should make them private too

**Review by mx-psi — COMMENTED** [review]
Thanks!! Just a couple more that I missed and we can merge this :)

**Review by mx-psi — APPROVED** [review]

**Pull request merged**